jetty-ee10/jetty-ee10-servlet/src/main/java/org/eclipse/jetty/ee10/servlet/DefaultServlet.java

```diff
@@ -1330,4 +886,44 @@ private enum WelcomeServletMode
          */
         EXACT
     }
+
+    private static class AsyncContextCallback implements Callback
+    {
+        private final AsyncContext _asyncContext;
+        private final HttpServletResponse _response;
+
+        private AsyncContextCallback(AsyncContext asyncContext, HttpServletResponse response)
+        {
+            _asyncContext = asyncContext;
+            _response = response;
+        }
+
+        @Override
+        public void succeeded()
+        {
+            _asyncContext.complete();
+        }
+
+        @Override
+        public void failed(Throwable x)
+        {
+            try
+            {
+                if (LOG.isDebugEnabled())
+                    LOG.debug("AsyncContextCallback failed {}", _asyncContext, x);
+                _response.sendError(-1);
+            }
+            catch (IOException e)
+            {
+                if (ExceptionUtil.areNotAssociated(x, e))
```

Comment:
This pattern became quite common in the 12 branch, I now wonder why it wasn't written as `ExceptionUtil.safeAddSuppressed(x, e)` instead?

Reply:
OK I did it